grid2op/Rules/BaseRules.py

```diff
@@ -20,6 +20,17 @@ class BaseRules(ABC):
     In :class:`grid2op.Environment`, only action of the users are checked for legality.
 
     """
+    
+    def initialize(self, env):
+        """
+        This function is used to inform the class instance about the environment specification. 
+        It can be the place to assert the defined rules are suited for the environement.
+        Parameters
+        ----------
+        env: :class:`grid2op.Environment.Environment`
+            The environment on which the action is performed. 
+        """
```

Comment:
Maybe just say explicitly that the env is not totally initialized when this is called.

Reply:
ok !